### PR TITLE
Update hooks types

### PR DESCRIPTION
**Pull request by oTkPoBeHuE**

### CODE REVIEW

File: packages/urban-bot-facebook/src/UrbanBotFacebook.ts [hunk 1]
```diff
@@ -71,9 +71,9 @@ export class UrbanBotFacebook implements UrbanBot<FacebookBotMeta> {
         expressApp.use('/facebook/*', json({ verify: this.verifyRequestSignature }));
 
         expressApp.get('/facebook/webhook', (req, res) => {
-            const mode = req.query['hub.mode'];
-            const token = req.query['hub.verify_token'];
-            const challenge = req.query['hub.challenge'];
+            const mode = req.query?.['hub.mode'];
```
Comment (oTkPoBeHuE) [inline packages/urban-bot-facebook/src/UrbanBotFacebook.ts:74]: Maybe it is not necessary

File: packages/urban-bot/src/components/Router/Router.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ export function Router({ children, withInitializeCommands = false }: RouterProps
         if (childrenArray.some(matchChild('/'))) {
             navigate('/');
         }
-    }, []);
+    }, []); // TODO: childrenArray ?
```
Comment (oTkPoBeHuE) [inline packages/urban-bot/src/components/Router/Router.tsx]: Maybe we forgot childrenArray ?

File: packages/urban-bot/src/components/Root.tsx [hunk 3]
```diff
@@ -76,14 +75,15 @@ export function Root<Bot extends UrbanBot>({
...
+        function handler(message: UrbanSyntheticEvent<Metadata>) {
             const { chat } = message;
             const { id: chatId } = chat;
 
+            // TODO:  <Metadata> is excessive
             if (!chatsRef.current.has(chatId)) {
                 chatsRef.current.set(
                     chat.id,
-                    <Chat<Bot>
+                    <Chat<Metadata>
```
Comment (oTkPoBeHuE) [inline packages/urban-bot/src/components/Root.tsx]: is **Metadata** not necessary?

File: packages/urban-bot/src/node.ts [hunk 4]
```diff
@@ -52,13 +48,16 @@ export function createNode<Bot extends UrbanBot>(
...
-export function appendChildNode<Bot extends UrbanBot>(
-    parentNode: UrbanNode<Bot> | UrbanNodeRoot,
-    childNode: UrbanNode<Bot>,
+export function appendChildNode<Metadata extends UrbanBotMeta>(
+    parentNode: UrbanNode<Metadata> | UrbanNodeRoot,
+    childNode: UrbanNode<Metadata>,
 ) {
+    // TODO: Fix me UrbanNodeRoot.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
```
Comment (oTkPoBeHuE) [inline packages/urban-bot/src/node.ts:60]: **childNodes** is
``` ts
UrbanNode<Metadata>
``` 
but  **parentNode** is  
 ``` ts
 UrbanNode<Metadata> | UrbanNodeRoot
```

File: packages/urban-bot/src/components/Router/Router.tsx (hunk 2 above)
Comment (ledamint) [reply packages/urban-bot/src/components/Router/Router.tsx]: Please add disable line

File: packages/urban-bot/src/components/Root.tsx (hunk 3 above)
Comment (ledamint) [reply packages/urban-bot/src/components/Root.tsx]: Maybe

File: packages/urban-bot/src/components/Router/Router.tsx (hunk 2 above)
Comment (ledamint) [reply packages/urban-bot/src/components/Router/Router.tsx]: please add TODO
// TODO check router children rerendering

File: packages/urban-bot/src/node.ts (hunk 4 above)
Comment (ledamint) [reply packages/urban-bot/src/node.ts:60]: // FIXME UrbanNodeRoot
I'll take a look

**Comment by ledamint** [comment]
Please use naming `UrbanTelegramMeta`.
And add generic looks like
`export type TelegramBotMeta<Payload = TelegramPayload> = {`

File: packages/urban-bot/src/components/Router/Router.tsx (hunk 2 above)
Comment (oTkPoBeHuE) [reply packages/urban-bot/src/components/Router/Router.tsx]: Fixed

File: packages/urban-bot/src/components/Root.tsx (hunk 3 above)
Comment (oTkPoBeHuE) [reply packages/urban-bot/src/components/Root.tsx]: Deleted

File: packages/urban-bot/src/node.ts (hunk 4 above)
Comment (oTkPoBeHuE) [reply packages/urban-bot/src/node.ts:60]: Resolve

**Pull request merged**